### DISCUSSION TITLE
Changes to RAG context and document handling in 0.3.6?

### DISCUSSION THREAD

**Discussion by ProjectMoon** [open-webui/open-webui#3516] [General]
I've noticed a behavior change in 0.3.6 that I'm not sure is a bug or not. I kind of think it is. 

In versions prior to 0.3.6, activating the RAG context by uploading a file or via web search would always have the model search the context on any future responses. But in 0.3.6, it seems to only use the context in the immediate response to the message that triggered the context.

This is particularly noticeable when uploading a file. Your first response will use the file as context, but the rest ignore the file as if it was never uploaded.

Was this an intentional change in behavior?

**Comment by ProjectMoon** [comment]
So I'm still seeing this. If I upload a file, only the immediate reply answers questions about the file. How can I get subsequent replies to answer?

**Reply by tjbck** [reply]
Related: https://github.com/open-webui/open-webui/issues/3674#issuecomment-2211788052

The default behaviour has been updated to clear out the document from the context after the first message, however, we're planning on adding an option to toggle between the new/old behaviour for 0.3.10, stay tuned!

**Reply by ProjectMoon** [reply]
Cool. There are arguments for both ways of doing it. With the old behavior, it would sometimes get "stuck" on the context, not being able to answer anything outside of it, or expand beyond what it finds in the context. Changing the handling of the context to user prompt instead of system prompt made it a lot better.

But it also makes sense that you'd want to ask more questions about a given document. Some kind of smooth UX to toggle on/off would be good. Maybe a little message that pops up after first upload saying "If you want to continue asking questions about documents in this chat, press the button to turn the feature on" (worded better of course).

But for right now: there's no way to ask questions about the document beyond one message, right?

**Reply by justinh-rahb** [reply]
> But for right now: there's no way to ask questions about the document beyond one message, right?

If you add a document to a Model's Documents configuration it keeps it in the context for the whole chat with that model.

**Reply by ProjectMoon** [reply]
> > But for right now: there's no way to ask questions about the document beyond one message, right?
> 
> If you add a document to a Model's Documents configuration it keeps it in the context for the whole chat with that model.

That's true. That still works as before. I am more curious about the flow of uploading files and referencing documents, which seem to be cleared out after the immediate reply. Guess I'll be waiting for 0.3.10!
